File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        maxLen = 0
        if not s:
            return maxLen
        current_s = s[0]
        for i in range(1, len(s)):
            if s[i] in current_s:
                for j in range(len(current_s)):
                    if s[i] == current_s[j]:
                        current_s = current_s[j+1:]
                        break;
            current_s += s[i]
            maxLen = max(len(current_s), maxLen)
        return maxLen if maxLen != 0 else len(s)
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.length() == 0) {
            return 0;
        }
        // define two point
        int left = 0;
        int right = 1;
        int result = 1;
        while (left <= right && right < s.length()) {
            String sliced = s.substring(left, right);
            // without repeating characters
            char currentChar = s.charAt(right);
            if (sliced.indexOf(currentChar) == -1) {
                // update result to max value between size of sliced and current result
                result = Math.max(right-left+1, result);
                right++;
            } else {
                left++;
            }
        }

        return result;
    }
}